c2corg_api/views/waypoint.py

```diff
@@ -34,9 +33,96 @@ def get(self):
     def collection_post(self):
         waypoint = schema_waypoint.objectify(self.request.validated)
 
+        # TODO additional validation: at least one culture, only one instance
+        # for each culture
+
         DBSession.add(waypoint)
         DBSession.flush()
 
         self._create_new_version(waypoint)
 
         return to_json_dict(waypoint, schema_waypoint)
+
+    @view(schema=schema_update_waypoint, validators=validate_id)
+    def put(self):
+        id = self.request.validated['id']
+        waypoint_in = \
+            schema_waypoint.objectify(self.request.validated['document'])
+        self._check_document_id(id, waypoint_in.document_id)
+
+        waypoint = self._get_waypoint(id)
+        self._check_versions(waypoint, waypoint_in)
+        old_versions = waypoint.get_versions()
+        waypoint.update(waypoint_in)
+
+        DBSession.merge(waypoint)
+        DBSession.flush()
```

Comment:
So the flush operation will cause an SQLAlchemy `StaleDataError` if the versions don't match. Don't you want to convert this to an `HTTPConflict` error?